### PR TITLE
Use os.Exit() for ReturnStmt in main(). Fixes #79

### LINKED ISSUE #79: Correctly handle the exit code returned from main()
The `main` function in Go does not return a value. c2go will discard the return value so the program will always finish with a successful exit code.

We need to recognise the `return` in the `main` function and call the system exit so that the correct status is passed back to the calling process.

Search for "issues/79" in the codebase.

### PR DESCRIPTION
**Pull request by x-tyger**
Working on using os.Exit() to return from main.

This one might need a little work, I'm curious of your thoughts.

---
This change is [Reviewable](https://reviewable.io/reviews/elliotchance/c2go/97)

### CODE REVIEW

File: transpiler/functions.go [hunk 1]
```diff
@@ -210,11 +210,17 @@ func transpileReturnStmt(n *ast.ReturnStmt, p *program.Program) (*goast.ReturnSt
...
-	// main() function is not allow to return a result.
-	//
-	// TODO: Correctly handle the exit code returned from main()
-	// https://github.com/elliotchance/c2go/issues/79
+	// main() function is not allow to return a result. Use os.Exit if non-zero
 	if p.FunctionName == "main" {
+		litExpr, isLiteral := e.(*goast.BasicLit)
+		if !isLiteral || (isLiteral && litExpr.Value != "0") {
+			p.AddImport("os")
+			e := &goast.CallExpr{
```
Comment (cristim) [inline transpiler/functions.go]: I'd just put this inline in the return statement

File: transpiler/functions.go [hunk 2]
```diff
@@ -210,11 +210,17 @@ func transpileReturnStmt(n *ast.ReturnStmt, p *program.Program) (*goast.ReturnSt
 
 	results := []goast.Expr{types.CastExpr(p, e, eType, f.ReturnType)}
 
-	// main() function is not allow to return a result.
-	//
-	// TODO: Correctly handle the exit code returned from main()
-	// https://github.com/elliotchance/c2go/issues/79
+	// main() function is not allow to return a result. Use os.Exit if non-zero
```
Comment (cristim) [inline transpiler/functions.go]: I think you meant "allowed"

File: transpiler/functions.go (hunk 1 above)
Comment (x-tyger) [inline transpiler/functions.go]: Done.

File: transpiler/functions.go (hunk 2 above)
Comment (x-tyger) [inline transpiler/functions.go]: Done.

**Comment by cristim** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/elliotchance/c2go/97#-:-KjnMBqRcZmITQt5CoBn:bnfp4nl)*

**Comment by cristim** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/elliotchance/c2go/97#-:-KjnNenKAq9w6L-lfs5V:bnfp4nl)*

**Pull request merged**